pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -1176,84 +1188,1309 @@ func Test_server_Insert(t *testing.T) {
+						Name: name,
+						Uuid: id,
+						Ips:  []string{ip},
+					},
+				},
+			}
+		}(),
+		func() test {
+			dim := 3
+
+			return test{
+				name: "Boundary Value Testing case 6: Insert nil insert request fail",
+				args: args{
+					ctx: ctx,
+					req: nil,
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    dim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Float.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				want: want{
+					// IncompatibleDimensionSize error will be returned
+					err: func() error {
+						err := errors.ErrIncompatibleDimensionSize(0, 3)
+						return status.WrapWithInvalidArgument("Insert API Incompatible Dimension Size detected",
+							err,
+							&errdetails.RequestInfo{
+								RequestId:   "",
+								ServingData: errdetails.Serialize(nil),
+							},
+							&errdetails.BadRequest{
+								FieldViolations: []*errdetails.BadRequestFieldViolation{
+									{
+										Field:       "vector dimension size",
+										Description: err.Error(),
+									},
+								},
+							},
+							&errdetails.ResourceInfo{
+								ResourceType: ngtResourceType + "/ngt.Insert",
+								ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+							})
+					}(),
+				},
+			}
+		}(),
+		func() test {
+			id := "1"
+			dim := 3
+
+			return test{
+				name: "Boundary Value Testing case 7: Insert nil vector fail",
+				args: args{
+					ctx: ctx,
+					req: &payload.Insert_Request{
+						Vector: &payload.Object_Vector{
+							Id:     id,
+							Vector: nil,
+						},
+					},
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    dim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Float.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				want: want{
+					// IncompatibleDimensionSize error will be returned
+					err: func() error {
+						err := errors.ErrIncompatibleDimensionSize(0, 3)
+						return status.WrapWithInvalidArgument("Insert API Incompatible Dimension Size detected",
+							err,
+							&errdetails.RequestInfo{
+								RequestId:   "1",
+								ServingData: errdetails.Serialize(nil),
+							},
+							&errdetails.BadRequest{
+								FieldViolations: []*errdetails.BadRequestFieldViolation{
+									{
+										Field:       "vector dimension size",
+										Description: err.Error(),
+									},
+								},
+							},
+							&errdetails.ResourceInfo{
+								ResourceType: ngtResourceType + "/ngt.Insert",
+								ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+							})
+					}(),
+				},
+			}
+		}(),
+		func() test {
+			id := "1"
```

Comment:
could you please define more meaningful value if you'd like to define directory? :sweat_smile:
or use some random generator?

Reply:
I will define it like `id :="uuid 1"`

Reply:
`uuid-1` is better.

Reply:
fixed